[IMP] web_editor: clean media link tool

Guillaume-gdi opened the pull request:
As we added the media link tool in a stable release (15.0) with [1], we
had to make sure that there would be no cache issues.
We don't have to worry about that for the next version so the code is
simplified with this commit.

[1] odoo#79295
--
task-2666467

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/78578).

Guillaume-gdi commented:
Since web_editor does not have website as a dependency but website does have web_editor as a dependency, I have moved the utility functions I need from website to web_editor.
Obviously I refactored the elements that also used these utils functions. :)

Guillaume-gdi commented on addons/web_editor/static/src/js/wysiwyg/wysiwyg.js:
```diff
@@ -818,6 +818,18 @@ const Wysiwyg = Widget.extend({
+     * Activates the link tool for the image.
+     */
+    toggleLinkOnImage() {
+        const buttonEl = document.querySelectorAll('we-button[data-set-link]')[1];
+        if (!buttonEl.classList.contains('active')) {
+            buttonEl.click();
+        }
+        setTimeout(function () { // Wait for the URL input to appear.
+            document.querySelectorAll('we-input[data-set-url]')[1].querySelector('input').focus();
```
I know it's not ideal to search for an item from the document but I haven't found any other way to access the editor's items from this file.

qsm-odoo reviewed (approved):
I updated myself but FYI:

- When you make a reference with `[XXX]`, at the bottom it needs to be `[XXX]: YYY` (with the `:`) which is the proper way to make a markdown reference (it does not have any effects in commit messages but it does in github message (like PR description)). Just a nice to have.

- About the reference (YYY), I changed the PR link to the merged commit link. I prefer to link to commits than PR in general when it makes sense. It's one less step for readers and it allows to see the commit diff from the hash outside of github.

- "As we added the media link tool in a stable release (15.0) with [1], we had to make sure that there would be no cache issues." -> It has nothing to do with cache issues. In stable, users would have received your JS changes but not your XML changes as they would require a *database* update (-u web_editor) which is not done in stable.

@robodoo r+
